backend/src/apiserver/server/pipeline_server.go

```diff
@@ -98,427 +108,1027 @@ type PipelineServer struct {
+	if pipelineId == "" {
+		return nil, util.NewInvalidInputError("[PipelineServer %s]: Failed to get the latest pipeline version as pipeline id is empty.", s.options.ApiVersion)
 	}
+	// Check authorization
 	resourceAttributes := &authorizationv1.ResourceAttributes{
 		Verb: common.RbacResourceVerbGet,
 	}
-	err = s.CanAccessPipeline(ctx, request.Id, resourceAttributes)
-	if err != nil {
-		return nil, util.Wrap(err, "Failed to authorize the requests.")
+	if err := s.canAccessPipeline(ctx, pipelineId, resourceAttributes); err != nil {
+		return nil, util.Wrap(err, fmt.Sprintf("[PipelineServer %s]: Failed to get the latest pipeline version due authorization error for pipeline id %v.", s.options.ApiVersion, pipelineId))
 	}
-	return ToApiPipeline(pipeline), nil
+	return s.resourceManager.GetLatestPipelineVersion(pipelineId)
 }
 
-func (s *PipelineServer) GetPipelineByNameV1(ctx context.Context, request *api.GetPipelineByNameRequest) (*api.Pipeline, error) {
-	if s.options.CollectMetrics {
-		getPipelineRequests.Inc()
+// Fetches pipeline and (optionally) pipeline version for a given name and namespace.
+func (s *PipelineServer) getPipelineByName(ctx context.Context, name string, namespace string, apiRequestVersion string) (*model.Pipeline, *model.PipelineVersion, error) {
+	//If namespace is "-" transform it to an empty string ""
+	if namespace == model.NoNamespace {
+		namespace = ""
+	}
+	if namespace != "" && namespace != s.options.DefaultNamespace && common.IsMultiUserMode() {
+		resourceAttributes := &authorizationv1.ResourceAttributes{
+			Namespace: namespace,
+			Verb:      common.RbacResourceVerbGet,
+		}
+		if err := s.haveAccess(ctx, resourceAttributes); err != nil {
+			return nil, nil, util.Wrap(err, fmt.Sprintf("[PipelineServer %s]: Failed to fetch a pipeline due to authorization error. Check if you have read permission to namespace %v.", s.options.ApiVersion, namespace))
+		}
 	}
+	if apiRequestVersion == "v1beta1" {
+		return s.resourceManager.GetPipelineByNameAndNamespaceV1(name, namespace)
+	} else if apiRequestVersion == "v2beta1" {
+		p, err := s.resourceManager.GetPipelineByNameAndNamespace(name, namespace)
+		return p, nil, err
+	}
+	return nil, nil, util.NewInternalServerError(
+		errors.New("Wrong api version detected."),
+		"[PipelineServer %s]: Failed to get a pipeline by name and namespace. API request version %v. Please, file a bug on github: https://github.com/kubeflow/pipelines/issues.",
+		s.options.ApiVersion,
+		apiRequestVersion,
+	)
+}
 
-	//If namespace is "-" transform it to an emtpy string ""
-	if request.Namespace == model.NoNamespace {
-		request.Namespace = ""
+// Fetches an array of []model.Pipeline and an array of []model.PipelineVersion for given search query parameters.
+func (s *PipelineServer) listPipelines(ctx context.Context, namespace string, pageToken string, pageSize int32, sortBy string, filter string, apiRequestVersion string) ([]*model.Pipeline, []*model.PipelineVersion, int, string, error) {
```

Comment:
Same as above, since this method should be accessible from other packages, it should start with an upper case.

Reply:
This function is not called outside of `PipelineServer`